### ISSUE TITLE
wrong logrotate configuration in debian package at packagecloud.io

### ISSUE THREAD

**Issue by olivluca** [netdata/netdata#8898]
##### Bug report summary

If you install the debian package from packagecloud.io, the logs aren't reopened so netdata will still append to the old log file.

##### OS / Environment

```
debian 10 (buster) x86_64
```

##### Netdata version

 v1.21.1

```
$ LC_ALL=C dpkg -l netdata
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-======================================
ii  netdata        1.21.1       amd64        real-time charts for system monitoring

```

##### Steps To Reproduce

1. install the debian package from packagecloud.io
2. start it
3. wait one day and see that /var/log/netdata/error.log is empty and messages are appended to /var/log/netdata/error.log.1

##### Expected behavior

after the logs have been rotated the new messages should be appended to /var/log/netdata/error.log

##### Cause

the logrotate configuration supplied with the package (/etc/logrotate.d/netdata) uses 

``` /bin/kill -HUP `cat /var/run/netdata/netdata.pid 2>/dev/null` 2>/dev/null || true```

in the postrotate section, however the pid file doesn't exist since the systemd unit supplied doesn't use the -P parameter to start netdata (it is in the init.d file but that's not used with systemd).

```
$ systemctl status netdata
● netdata.service - Real time performance monitoring
   Loaded: loaded (/lib/systemd/system/netdata.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2020-05-04 14:59:56 CEST; 18h ago
 Main PID: 14984 (netdata)
    Tasks: 35 (limit: 2358)
   Memory: 133.1M
   CGroup: /system.slice/netdata.service
           ├─14984 /usr/sbin/netdata -D
           ├─15077 /usr/libexec/netdata/plugins.d/go.d.plugin 1
           ├─15082 /usr/bin/python /usr/libexec/netdata/plugins.d/python.d.plugin 1
           ├─25886 /usr/libexec/netdata/plugins.d/nfacct.plugin 1
           ├─25887 /usr/libexec/netdata/plugins.d/apps.plugin 1
           └─31995 bash /usr/libexec/netdata/plugins.d/tc-qos-helper.sh 1
```
A possible solution would be to use `netdatacli reopen-logs` instead of  `kill -HUP` as suggested in issue #8244

The files are as distributed in the package:

```
$ dpkg -S /etc/logrotate.d/netdata /lib/systemd/system/netdata.service 
netdata: /etc/logrotate.d/netdata
netdata: /lib/systemd/system/netdata.service
$ sudo dpkg -V netdata
??5?????? c /etc/netdata/netdata.conf

```

**Comment by vince-lessbits** [comment]
This also happens on Ubuntu 18.04.

**Comment by prologic** [comment]
@olivluca Would you be willing to put up a PR that implements:

> A possible solution would be to use netdatacli reopen-logs instead of kill -HUP as suggested in issue #8244

I will happily review this. I would prefer we use the `netdatacli` tool as this talks directly to the running daemon over its IPC and _hopefully_ something we can rely on that isn't system-dependent per se.

Thanks!

**Comment by olivluca** [comment]
I created the pull request, but I'm checking the machines where I applied that change when I originally reported this and error.log is empty and the messages are still appended to error.log.1. If I invoke netdatacli manually the logs are reopened.

**Comment by olivluca** [comment]
OK, all the machines where the new logrotate configuration doesn't work are debian 10, on mandriva and ubuntu 18.04 the logs are rotated. I guess the problem lies in debian 10.

**Comment by prologic** [comment]
Thats kind of odd. And `netdatacli` manually works on those Debian 10 boxen? Any errors reported from the logrotate side from execve's the `netdatacli`?

**Comment by olivluca** [comment]
Yes, as I said invoking netdatacli manually worked. Now I see this error in the syslog of those machines:

```
Jun 18 00:00:51 dc1 logrotate[23563]: uv_pipe_connect(): no such file or directory
Jun 18 00:00:51 dc1 logrotate[23563]: Make sure the netdata service is running.
Jun 18 00:00:51 dc1 logrotate[23563]: error: error running shared postrotate script for '/var/log/netdata/*.log '

```

**Comment by olivluca** [comment]
OK, in debian 10 the systemd unit for logrotate has

`PrivateTmp=true`

hence netdatacli cannot connect to the socket.
Maybe it's better to leave logrotate as it is and modify the netdata systemd unit to write the pid **or** put the socket somewhere else (e.g. in /run, I don't know if systemd also messes with that)

**Comment by olivluca** [comment]
Mmmg, the pipe location is hardcoded in daemon/commands.h so I guess the only option is to modify the systemd unit in the package to write the pid in /var/run/netdata/netdata.pid

**Comment by prologic** [comment]
@netdata/agent Can we add some command-line flags to make this a bit more convenient so we can support some of the more esoteric feature of SystemD :D

**Comment by Ferroin** [comment]
> @netdata/agent Can we add some command-line flags to make this a bit more convenient so we can support some of the more esoteric feature of SystemD :D

I'd argue that we should actually fix the pipe location, as discussed in #7587.

**Comment by Ferroin** [comment]
A number of things tangentially related to this have changed, so if someone could check and confirm if it’s still an issue that would be appreciated.

**Comment by nodiscc** [comment]
I can still reproduce this (Netdata 1.33.1 package from packagecloud.io, Debian 11). `/var/log/netdata/error.log` is empty and messages are appended to `/var/log/netdata/error.log.1`
